kong/plugins/http-log/handler.lua

```diff
@@ -189,12 +189,47 @@ function HttpLogHandler:log(conf)
 
   local queue_conf = Queue.get_plugin_params("http-log", conf, make_queue_name(conf))
   kong.log.debug("Queue name automatically configured based on configuration parameters to: ", queue_conf.name)
+  local entry = cjson.encode(kong.log.serialize())
+
+  if queue_conf.max_batch_size == 1 then
+    local ok ,err = ngx.timer.at(0, function()
+      local retry_count = 0
+      local start_time = ngx.now()
+      while true do
+        -- fixme: optmize the log as these may not appropriate at here
+        kong.log.debug("passing 1 entries to handler")
+        local ok, err = send_entries(conf, { entry }) -- fixme: should we pcall?
+        if ok then
+          kong.log.debug("handler processed 1 entries successfully")
```

Comment:
Again, I don't think the log message is needed.  It'd be better to instrument send_entries.

Reply:
I deleted this line. Could you expand a little about "instrument send_entries"?

Reply:
If debugging information is needed, `send_entries` should log the number of entries that it is sending so that the message is shown even when queuing is enabled.

Reply:
The func `send_entries` currently has two args, and it is also the handler called inside the queue. Changing that requires more code effect -- we break the original semantics?